adapters/repos/db/lsmkv/memtable_test.go

```diff
@@ -0,0 +1,59 @@
+package lsmkv
+
+import (
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// This test prevents a regression on
+// https://www.youtube.com/watch?v=OS8taasZl8k
+func Test_MemtableSecondaryKeyBug(t *testing.T) {
+	dir := t.TempDir()
+	m, err := newMemtable(path.Join(dir, "will-never-flush"), StrategyReplace, 1, nil)
+	require.Nil(t, err)
+
+	t.Run("add initial value", func(t *testing.T) {
+		err = m.put([]byte("my-key"), []byte("my-value"),
+			WithSecondaryKey(0, []byte("secondary-key-initial")))
+		require.Nil(t, err)
+	})
```

Comment:
Why is adding a value as extra test?

Reply:
The idea was to have the blocks as small as possible so that reading the title reads like a description of going on. I could also merge this with the next block. WDYT?

Reply:
I would remove the t.run() around it, there is not really anything tested or?

My main annoyence with this is that when the 5th test fails, I cannot run just the 5th test, but I need to run first the "add test" and then the 5th test (or the whole class). I think if all "setup" operations are part of the class itself, they are automatically run in case I just run one test on its own.

Reply:
In this particular test suite you would need to run them all together anyway because each step does a specific modification to the state and then the next step would assert based on previous state. It's basically one big tests that is only split up into `t.run()`s for better readability.